bin/parse_config.py

```diff
@@ -29,73 +43,138 @@ def parse_config(args):
+        # This is mixing old and new manners of doing this quite a bit
+        # It will work for a PoC, but going forward it would be good to
+        # standardize.
+        if context.config['parallel']:
+            option_string += ' -parallel '
+            # grab number of cpus from host.
+            max_cpus = mp.cpu_count()
+            # use of -parallel defaults to cpu_count unless specified
+            if context.config.get('n_cpus'):
+                # However, we will do strict checking on max_cpus
+                if context.config['n_cpus'] <= max_cpus:
+                    option_string += '-openmp ' + str(context.config['n_cpus'])
+                else:
+                    option_string += '-openmp ' + str(max_cpus)
+            else:
+                option_string += '-openmp ' + str(max_cpus)
+
+        print(option_string)
 
     # Print options for recon-all
     if args.r:
-        print config['config']['register_surfaces']
+        print(config['config']['register_surfaces'])
 
     # Convert surfaces to obj
     if args.s:
-        print config['config']['convert_surfaces']
+        print(config['config']['convert_surfaces'])
 
     # Convert mgz to nifti
     if args.n:
-        print config['config']['convert_volumes']
+        print(config['config']['convert_volumes'])
 
     # Convert aseg stats to csv
     if args.a:
-        print config['config']['convert_stats']
+        print(config['config']['convert_stats'])
 
     # Process hippocampal subfields
     if args.c:
-        print config['config']['hippocampal_subfields']
+        print(config['config']['hippocampal_subfields'])
 
     # Process brainstem substructures
     if args.b:
-        print config['config']['brainstem_structures']
+        print(config['config']['brainstem_structures'])
 
     # Get subject code from archive input
     if args.z:
         try:
-            zip = zipfile.ZipFile(config['inputs']['anatomical']['location']['path'])
-            print zip.namelist()[0].split('/')[0]
+            zip = zipfile.ZipFile(
+                config['inputs']['anatomical']['location']['path'])
+            print(zip.namelist()[0].split('/')[0])
         except:
-            print ''
+            print('')
 
     # Parse config for license elements
     if args.l:
-        if not config['config'].has_key('freesurfer_license'):
-            print ""
+        # This will look for project level and input level freesurfer license
+        # files
+
+        # grab input, config, and project in order they are checked
+        # from input:
+        fs_license_file = context.get_input_path('freesurfer_license_file')
+        # from config:
+        fs_license = context.config.get('freesurfer_license')
+        # from project (needs api-key in config.json/manifest.json):
+        project_id = fw.get_analysis(destination_id).parents.project
+        project = fw.get_project(project_id)
+        project_license = project.get_file('license.txt')
```

Comment:
This is a completely sensible option, however it's not one of the [three "accepted" ways of providing a license](https://docs.flywheel.io/hc/en-us/articles/360013235453-How-to-include-a-Freesurfer-license-file-in-order-to-run-a-Freesurfer-or-fMRIPrep-gear-) (though it really should be). 
The third option is to provide the license as project context, or `info` on the project, and grab it from there: ```project.info.get('FREESURFER_LICENSE')```. Full example [here](https://github.com/flywheel-apps/fmriprep/blob/5.6.1_1.1.4/fs_license.py#L30-L36).

Reply:
I will make it an option.